spec/unit/matrix-client.spec.ts

```diff
@@ -268,6 +304,130 @@ describe("MatrixClient", function() {
         client.stopClient();
     });
 
+    describe('timestampToEvent', () => {
+        const roomId = '!room:server.org';
+        const eventId = "$eventId:example.org";
+        const unstableMSC3030Prefix = "/_matrix/client/unstable/org.matrix.msc3030";
+
+        it('should call stable endpoint', async () => {
+            httpLookups = [{
+                method: "GET",
+                path: `/rooms/${encodeURIComponent(roomId)}/timestamp_to_event`,
+                data: { event_id: eventId },
+                expectQueryParams: {
+                    ts: '0',
+                    dir: 'f',
+                },
+            }];
+
+            await client.timestampToEvent(roomId, 0, Direction.Forward);
+
+            expect(mocked(client.http.authedRequest).mock.calls.length).toStrictEqual(1);
+            const [method, path, queryParams,, { prefix }] = mocked(client.http.authedRequest).mock.calls[0];
```

Comment:
What's a good way to solve this Typescript lint?

https://github.com/matrix-org/matrix-js-sdk/actions/runs/3653310171/jobs/6172633351#step:5:7
```
Property 'prefix' does not exist on type 'IRequestOpts | undefined'.
```

This seems to work but is there something better?
```suggestion
            const [method, path, queryParams,, { prefix }  = { prefix: undefined }] = mocked(client.http.authedRequest).mock.calls[0];
```

Reply:
> @MadLittleMods when it is ready for re-review, please ping me oob and I'll take a look. The CI looks a bit angry at the moment.

@turt2live The CI is talking about this point. I've pushed the first change suggested here but still curious if anything comes to mind?

Reply:
@MadLittleMods was this resolved? Github appears to think it is.

Reply:
@turt2live The lint itself is resolved because I made the change as described above (see comment just above, "I've pushed the first change suggested here").

But I had left this unresolved ("but still curious if anything comes to mind?") in hopes there might be a better taste-making option here. This whole point is asking if there are better options.

It can be resolved if you have no further input to add.

Reply:
Hopefully you prefer the code after my little refactoring:

```
const [callMethod, callPath, callQueryParams, , callOpts] = call;
const callPrefix = callOpts?.prefix;
```